hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -610,21 +639,255 @@ def set_net_pay(self):
 			)
 		self.set_net_total_in_words()
 
+	def compute_taxable_earnings_for_year(self):
+		# get remaining numbers of sub-period (period for which one salary is processed)
+		self.remaining_sub_periods = get_period_factor(
+			self.employee, self.start_date, self.end_date, self.payroll_frequency, self.payroll_period
+		)[1]
+
+		# get taxable_earnings, opening_taxable_earning, paid_taxes for previous period
+		self.previous_taxable_earnings = self.get_taxable_earnings_for_prev_period(
+			self.payroll_period.start_date, self.start_date, self.tax_slab.allow_tax_exemption
+		)
+
+		# get taxable_earnings for current period (all days)
+		self.current_taxable_earnings = self.get_taxable_earnings(self.tax_slab.allow_tax_exemption)
+		self.future_structured_taxable_earnings = self.current_taxable_earnings.taxable_earnings * (
+			math.ceil(self.remaining_sub_periods) - 1
+		)
+
+		# get taxable_earnings, addition_earnings for current actual payment days
+		self.current_taxable_earnings_for_payment_days = self.get_taxable_earnings(
+			self.tax_slab.allow_tax_exemption, based_on_payment_days=1
+		)
+		self.current_structured_taxable_earnings = (
+			self.current_taxable_earnings_for_payment_days.taxable_earnings
+		)
+		self.current_additional_earnings = (
+			self.current_taxable_earnings_for_payment_days.additional_income
+		)
+		self.current_additional_earnings_with_full_tax = (
+			self.current_taxable_earnings_for_payment_days.additional_income_with_full_tax
+		)
+
+		# Deduct taxes forcefully for unsubmitted tax exemption proof and unclaimed benefits in the last period
+		if self.payroll_period.end_date <= getdate(self.end_date):
+			self.deduct_tax_for_unsubmitted_tax_exemption_proof = 1
+			self.deduct_tax_for_unclaimed_employee_benefits = 1
```

Comment:
`compute_taxable_earnings_for_year` is huge. Can we push this into a separate function?